src/centrality.rs

```diff
@@ -360,6 +360,136 @@ pub fn digraph_closeness_centrality(
     }
 }
 
+/// Compute the weighted closeness centrality of each node in the graph.
+///
+/// The weighted closeness centrality is an extension of the standard closeness
+/// centrality measure where edge weights represent connection strength rather
+/// than distance. To properly compute shortest paths, weights are inverted
+/// so that stronger connections correspond to shorter effective distances.
+/// The algorithm follows the method described by Newman (2001) in analyzing
+/// weighted graphs.[Newman]
+///
+/// The edges originally represent connection strength between nodes.
+/// The idea is that if two nodes have a strong connection, the computed
+/// distance between them should be small (shorter), and vice versa.
+/// Note that this assume that the graph is modelling a measure of
+/// connection strength (e.g. trust, collaboration, or similarity).
+/// If the graph is not modelling a measure of connection strength,
+/// the function `weight_fn` should invert the weights before calling this
+/// function, if not it is considered as a logical error.
+///
+/// In the case of a graphs with more than one connected component there is
+/// an alternative improved formula that calculates the closeness centrality
+/// as "a ratio of the fraction of actors in the group who are reachable, to
+/// the average distance".[WF]
+/// You can enable this by setting `wf_improved` to `true`.
+///
+/// :param PyGraph graph: The input graph. Can either be a
+///     :class:`~rustworkx.PyGraph` or :class:`~rustworkx.PyDiGraph`.
+/// :param weight_fn: An optional input callable that will be passed the edge's
+///    payload object and is expected to return a `float` weight for that edge.
+///    If this is not specified ``default_weight`` will be used as the weight
+///    for every edge in ``graph``
+/// :param bool wf_improved: This is optional; the default is True. If True,
+///   scale by the fraction of nodes reachable.
+/// :param float default_weight: If ``weight_fn`` is not set the default weight
+///     value to use for the weight of all edges
+///
+/// :returns: A dictionary mapping each node index to its closeness centrality.
+/// :rtype: CentralityMapping
+#[pyfunction(signature = (graph, weight_fn, wf_improved=true, default_weight = 0.0))]
+pub fn graph_newman_weighted_closeness_centrality(
+    py: Python,
+    graph: &graph::PyGraph,
+    weight_fn: PyObject,
+    wf_improved: bool,
+    default_weight: f64,
+) -> PyResult<CentralityMapping> {
+    let mut edge_weights = vec![default_weight; graph.graph.edge_bound()];
+    let cost_fn = CostFn::from(weight_fn);
+    for edge in graph.graph.edge_indices() {
```

Comment:
To support the `default_weight` argument, follow this instead:
https://github.com/Qiskit/rustworkx/blob/a9460dcdd045bfc8e5fdfc8e1d3749feea1a3f7b/src/centrality.rs#L724

`CostFn::try_from` will assign the default if `weight_fn` is None